cppcache/src/Log.cpp

```diff
@@ -502,178 +351,77 @@ LogLevel Log::charsToLevel(const std::string& chars) {
   }
 }
 
-char* Log::formatLogLine(char* buf, LogLevel level) {
-  if (g_pid == 0) {
-    g_pid = boost::this_process::get_id();
-    ACE_OS::uname(&g_uname);
-  }
+std::string Log::formatLogLine(LogLevel level) {
+  std::stringstream msg;
   const size_t MINBUFSIZE = 128;
   auto now = std::chrono::system_clock::now();
   auto secs = std::chrono::system_clock::to_time_t(now);
   auto microseconds = std::chrono::duration_cast<std::chrono::microseconds>(
       now - std::chrono::system_clock::from_time_t(secs));
   auto tm_val = apache::geode::util::chrono::localtime(secs);
-  auto pbuf = buf;
-  pbuf += std::snprintf(pbuf, 15, "[%s ", Log::levelToChars(level));
-  pbuf += std::strftime(pbuf, MINBUFSIZE, "%Y/%m/%d %H:%M:%S", &tm_val);
-  pbuf += std::snprintf(pbuf, 15, ".%06" PRId64 " ",
-                        static_cast<int64_t>(microseconds.count()));
-  pbuf += std::strftime(pbuf, MINBUFSIZE, "%Z ", &tm_val);
+  msg << "[" << Log::levelToChars(level) << " ";
+  char timebuf[MINBUFSIZE];
 
-  std::snprintf(pbuf, 300, "%s:%d %" PRIu64 "] ", g_uname.nodename, g_pid,
-                hacks::aceThreadId(ACE_OS::thr_self()));
+  std::strftime(timebuf, MINBUFSIZE, "%Y/%m/%d %H:%M:%S", &tm_val);
+  msg << timebuf;
 
-  return buf;
-}
+  std::snprintf(timebuf, 15, ".%06" PRId64 " ",
+                static_cast<int64_t>(microseconds.count()));
+  msg << timebuf << " ";
 
-void Log::put(LogLevel level, const std::string& msg) {
-  put(level, msg.c_str());
+  std::strftime(timebuf, MINBUFSIZE, "%Z ", &tm_val);
+  msg << timebuf << " " << boost::asio::ip::host_name() << ":"
```

Comment:
I am a bit concerned this may result in a system call the resolver for every log message. The previous code cached this. I suggest a `::hostname()` method with a static resolved hostname using this method (Meyer's Singleton). The likely hood of the hostname changing over the life of a client, while not 0, is pretty darn close to 0.

Reply:
Cached g_hostName string.  Didn't read the fine print here and implement an accessor, but I think we're okay as is.